src/main/java/org/tugraz/sysds/runtime/instructions/cp/VariableCPInstruction.java

```diff
@@ -333,7 +330,18 @@ else if ( voc == VariableOperationCode.Write ) {
 		case CreateVariable:
 			// variable name
 			DataType dt = DataType.valueOf(parts[4]);
-			ValueType vt = dt==DataType.MATRIX ? ValueType.FP64 : ValueType.STRING;
```

Comment:
Other than the TODO, how is the below code different?

Reply:
It is not, I just thought it might show that more work is necessary. I reverted it.